jrnl/plugins/meta.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+import importlib
+import pkgutil
+
+import jrnl.contrib.exporter
+import jrnl.contrib.importer
+import jrnl.plugins.exporter
+import jrnl.plugins.importer
+
+from .template_exporter import __all__ as template_exporters
+
+__exporters_builtin = list(
+    pkgutil.iter_modules(
+        jrnl.plugins.exporter.__path__, jrnl.plugins.exporter.__name__ + "."
+    )
+)
+__exporters_contrib = list(
+    pkgutil.iter_modules(
+        jrnl.contrib.exporter.__path__, jrnl.contrib.exporter.__name__ + "."
+    )
+)
+
+__importers_builtin = list(
+    pkgutil.iter_modules(
+        jrnl.plugins.importer.__path__, jrnl.plugins.importer.__name__ + "."
+    )
+)
+__importers_contrib = list(
+    pkgutil.iter_modules(
+        jrnl.contrib.importer.__path__, jrnl.contrib.importer.__name__ + "."
+    )
+)
+
+__exporter_types_builtin = {
+    name: importlib.import_module(plugin.name)
+    for plugin in __exporters_builtin
+    for name in importlib.import_module(plugin.name).Exporter.names
+}
+__exporter_types_contrib = {
+    name: importlib.import_module(plugin.name)
+    for plugin in __exporters_contrib
+    for name in importlib.import_module(plugin.name).Exporter.names
+}
+__exporter_types_template = {
+    name: plugin for plugin in template_exporters for name in plugin.names
+}
+
+
+__importer_types_builtin = {
+    name: importlib.import_module(plugin.name)
+    for plugin in __importers_builtin
+    for name in importlib.import_module(plugin.name).Importer.names
+}
+__importer_types_contrib = {
+    name: importlib.import_module(plugin.name)
+    for plugin in __importers_contrib
+    for name in importlib.import_module(plugin.name).Importer.names
+}
+
+__exporter_types = {
+    **__exporter_types_builtin,
+    **__exporter_types_contrib,
+    **__exporter_types_template,
+}
```

Comment:
So, this would override builtin plugins with contrib plugins of the same type, right? Does that behavior get communicated to the user in any way? If somebody runs `jrnl --format markdown`, do they have any way to know if that's a builtin, or some random plugin they forgot they installed?

This would also be relevant for users reporting bugs. We need a way for us to know which version of which plugin they are using, I think.

Reply:
My first thought was to add the list of plugins (with versions) to the information displayed by `jrnl --version`. But nothing has been implemented quite yet.

Reply:
(Trying to do this from my phone wasn't working... so now I'm at the computer)

> this would override builtin plugins with contrib plugins of the same type, right?

Yes, that is what is supposed to happen. I haven't tested it yet....

> Does that behavior get communicated to the user in any way?

No, not currently. My thought was to extend `jrnl --version` to list the installed plugins, but to note the plugin used at the start of the export would be good too. In any case, the version of the plugin should be printed as well.